### ISSUE TITLE
Coach reports: Apply date rage to tabular report

### ISSUE THREAD

**Issue by harishbalachandran** [learningequality/ka-lite#5022]
**Detailed description:**
For the date range selected total exercise attempted is shown as 7, but the tabular column shows all the exercise of the all the kids instead of the selected date range.

**Expected behavior:**
Only the 7 exercise should be displayed on the column so that its easy for the teachers to see the progress made by the kids on a selected date range.

**Actual behavior:**
All the exercise are getting displayed for all the kids even though the selected date range shows total attempted exercise as 7. This is a huge problem for the teacher who has to scroll through all the exercise and is very time consuming. 
![screen shot 2016-03-21 at 6 44 31 pm](https://cloud.githubusercontent.com/assets/13453618/13919702/1c81fece-ef95-11e5-922a-e9ba26214df6.png)
![screen shot 2016-03-21 at 6 45 13 pm](https://cloud.githubusercontent.com/assets/13453618/13919704/1c8cb5ee-ef95-11e5-8d83-481f5d04f317.png)
![screen shot 2016-03-21 at 6 44 50 pm](https://cloud.githubusercontent.com/assets/13453618/13919703/1c86e420-ef95-11e5-8243-dfd80e1966b8.png)
![screen shot 2016-03-21 at 6 44 56 pm](https://cloud.githubusercontent.com/assets/13453618/13919706/1c96d57e-ef95-11e5-99f4-aaddf6339009.png)
![screen shot 2016-03-21 at 6 45 03 pm](https://cloud.githubusercontent.com/assets/13453618/13919705/1c91446a-ef95-11e5-985d-d4fe3ca409cc.png)

Please refer the screenshots for more details.

**Comment by sethachoi** [comment]
Currently true as of 0.16.x

Date filter does not filter which student's did the exercises/videos in the selected timespan. Instead, it just shows everyone who has done the exercise (date picker has no effect on who is shown on the tabular view)

**Comment by harishbalachandran** [comment]
I understand that the date filter does not have effect on the student list and all the students names are displayed row wise. 

But shouldn't date filter do the work of filtering out the topics attempted during that time period? 

In the screen shot above for the date filter 3/21/2016, there were only 7 exercises attempted, so shouldn't only those exercise attempted displayed on the column instead of all the exercises?

Please confirm my understanding.

**Comment by sethachoi** [comment]
Sorry if that was unclear, I was just confirming the bug was present on our 0.16 version rather than just being in the 0.15 version.

**Comment by harishbalachandran** [comment]
Thanks once again for the clarification :+1:

**Comment by rtibbles** [comment]
I think mostly this is an issue with generaterealdata rather than the coach reports - it seems to be generating way more exerciselogs than attemptlogs, hence giving the erroneous output.

It did however make me catch one small error with the date ranging for the tabular report.

**Comment by benjaoming** [comment]
@harishbalachandran are you around to triage if this is still present? `generaterealdata` has been fixed in #4985

**Comment by benjaoming** [comment]
In `kalite/coachreports/static/js/coachreports/tabular_reports/views.js`, we are correctly retrieving `start_date` and `end_date`, however it seems to be a different instance of `this.model` that isn't affected by events in `TimeSetView.set_range`.

```js
        this.data_model = new Models.CoachReportModel({
            facility: this.model.get("facility"),
            group: this.model.get("group"),
            start_date: date_string(this.model.get("start_date")),
            end_date: date_string(this.model.get("end_date")),
            topic_ids: this.model.get("topic_ids")
        });
```

**Comment by benjaoming** [comment]
Fixed in #5435
